src/bundle.rs

```diff
@@ -88,31 +157,59 @@ pub fn create_runtime_config(
                     ]));
                 mounts.push(m.clone());
             }
+            if !mounts.is_empty() {
+                if let Some(default_mounts) = spec.mounts() {
+                    mounts.extend(default_mounts.clone());
+                }
+                spec.set_mounts(Some(mounts));
+            }
         }
+    }
 
-        if !mounts.is_empty() {
-            if let Some(default_mounts) = spec.mounts() {
-                mounts.extend(default_mounts.clone());
+    if !labels.contains_key(ANNOTATION_OS) {
+        annotations.insert(ANNOTATION_OS.to_string(), image_config.os().to_string());
+    }
+    if !labels.contains_key(ANNOTATION_OS_VERSION) {
+        if let Some(version) = image_config.os_version() {
+            annotations.insert(ANNOTATION_OS_VERSION.to_string(), version.to_string());
+        }
+    }
+    if !labels.contains_key(ANNOTATION_OS_FEATURES) {
+        if let Some(features) = image_config.os_features() {
+            if let Ok(v) = serde_json::to_string(features) {
+                annotations.insert(ANNOTATION_OS_FEATURES.to_string(), v);
             }
-            spec.set_mounts(Some(mounts));
         }
-
-        if let Some(stop_signal) = config.stop_signal() {
-            annotations.insert(ANNOTATION_STOP_SIGNAL.to_string(), stop_signal.to_string());
+    }
+    if !labels.contains_key(ANNOTATION_ARCH) {
+        annotations.insert(
+            ANNOTATION_ARCH.to_string(),
+            image_config.architecture().to_string(),
+        );
+    }
+    if !labels.contains_key(ANNOTATION_VARIANT) {
+        if let Some(variant) = image_config.variant() {
+            annotations.insert(ANNOTATION_VARIANT.to_string(), variant.to_string());
         }
-        if let Some(exposed_ports) = config.exposed_ports() {
-            annotations.insert(
-                ANNOTATION_EXPOSED_PORTS.to_string(),
-                exposed_ports.join(","),
-            );
+    }
+    if !labels.contains_key(ANNOTATION_AUTHOR) {
+        if let Some(author) = image_config.author() {
+            annotations.insert(ANNOTATION_AUTHOR.to_string(), author.to_string());
+        }
+    }
+    if !labels.contains_key(ANNOTATION_CREATED) {
+        if let Some(created) = image_config.created() {
+            annotations.insert(ANNOTATION_CREATED.to_string(), created.to_string());
         }
     }
 
     spec.set_annotations(Some(annotations));
+    // TODO
     let bundle_config = bundle_path.join(BUNDLE_CONFIG);
     spec.save(&bundle_config)?;
 
-    Ok(bundle_config)
+    PinnedPathBuf::new(bundle_path, BUNDLE_CONFIG)
+        .context("The path of generated config.json file has changed, possible attack!")
```

Comment:
- Could you use `BUNDLE_CONFIG` rather than hard-coding `config.json`.
- I wonder if we want to use a standard "eye catcher" / tag for scenarios like this (maybe use a "SECURITY" prefix in the context message?) That would make errors like this more `grep(1)`-able.

Reply:
Any update on this?